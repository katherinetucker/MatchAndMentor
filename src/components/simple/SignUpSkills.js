import React from 'react';
import {
  Text,
  View,
  TextInput,
  Keyboard,
  Platform,
  KeyboardAvoidingView,
} from 'react-native';
import Button from './Button';
import ScreenHeader from './ScreenHeader';
import {
  updateSkillName,
  setProficiencyAndSaveSkill,
  submitSkills,
} from '../../actions';

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'space-between',
  },
  buttonStyle: {
    backgroundColor: '#55A838',
    margin: 20,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center',
  },
  textStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
  },
  inputStyle: {
    height: 40,
    borderBottomColor: 'black',
    borderBottomWidth: Platform.OS === 'ios' ? 1 : 0,
    margin: 20,
  },
  skillButtonStyle: {
    backgroundColor: '#333333',
    margin: 5,
    padding: 10,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  skillButtonTextStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 18,
    fontFamily: 'Montserrat-Regular',
  },
};

const SkillProficiencyBadge = props => (
  <View
    style={{
      width: 30,
      height: 30,
      backgroundColor: '#ffffff',
      borderRadius: 30,
      justifyContent: 'center',
      marginLeft: 10,
    }}>
    <Text
      style={{
        color: '#000000',
        alignSelf: 'center',
      }}>
      {props.proficiency}
    </Text>
  </View>
);

const SkillButton = props => (
  <Button style={styles.skillButtonStyle} onPress={() => null}>
    <Text style={styles.skillButtonTextStyle}>{props.name}</Text>
    <SkillProficiencyBadge proficiency={props.proficiency} />
  </Button>
);

const ProficiencyButton = props => (
  <Button
    title={props.proficiency}
    onPress={props.dispatch}
    style={{ ...styles.buttonStyle, flex: 0.2 }}>
    <Text style={styles.textStyle}>{props.proficiency}</Text>
  </Button>
);

function submit(dispatch) {
  Keyboard.dismiss();
  dispatch(submitSkills())
}

const SignUpSkills = props => {
  const proficiencyLevels = ['1', '2', '3', '4', '5'];
  return (
    <View style={styles.container}>
      <View>
        <ScreenHeader title={props.title} />
        <TextInput
          editable
          style={styles.inputStyle}
          maxLength={100}
          onChangeText={skillName => props.dispatch(updateSkillName(skillName))}
          onSubmitEditing={Keyboard.dismiss}
          onEndEditing={Keyboard.dismiss}
          placeholder={'Ex: JavaScript'}
          testID="skill-name-input"
        />
        <View
          style={{
            flex: 1,
            flexDirection: 'column',
            justifyContent: 'center',
          }}
        />
        <View style={{ flexDirection: 'row' }}>
          {proficiencyLevels.map(level => (
            <ProficiencyButton
              key={level}
              proficiency={level}
              dispatch={() => {
                props.dispatch(setProficiencyAndSaveSkill(level));
              }}
            />
          ))}
        </View>
      </View>
      <View style={{ flexDirection: 'row', flexWrap: 'wrap' }}>
        {props.skills.map(skill => (
          <SkillButton
            key={skill.name}
            name={skill.name}
            proficiency={skill.proficiency}
          />
        ))}
      </View>
      <KeyboardAvoidingView behavior="padding" keyboardVerticalOffset={64}>
        <Button
          style={
            props.skills !== null && props.skills.length > 0
              ? styles.buttonStyle
              : { ...styles.buttonStyle, backgroundColor: '#e9e9e9' }
          }
          onPress={() => submit(props.dispatch)}
          disabled={props.name === null || props.name === ''}
          testID="skills-submit-button">
          <Text style={styles.textStyle}>Continue</Text>
        </Button>
      </KeyboardAvoidingView>
    </View>
  );
};

export default SignUpSkills;
