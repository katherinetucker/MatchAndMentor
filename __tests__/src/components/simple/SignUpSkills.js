/* eslint-env jest */

import { Keyboard } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SignUpSkills from '../../../../src/components/simple/SignUpSkills';
import { findById } from '../../../../setupTests';

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native');
  return mockComponent('TouchableHighlight');
});

jest.mock('Keyboard', () => ({
  dismiss: jest.fn(),
  addListener: jest.fn(),
}));

const skill = {name: 'test', proficiency:1};
const skills = [skill];
const tree = renderer.create(<SignUpSkills dispatch={jest.fn()} skill={skill} skills={skills} />);

it('renders correctly', () => {
  expect(tree).toMatchSnapshot();
});

it('calls submit on button press', () => {
  const submitButton = findById(tree.toJSON(), 'skills-submit-button');
  submitButton.props.onPress();
  expect(Keyboard.dismiss).toBeCalled();
});

it('calls dispatch on change text', () => {
  const nameInput = findById(tree.toJSON(), 'skill-name-input');
  nameInput.props.onChangeText();
  expect(tree.toTree().props.dispatch).toBeCalled();
});
