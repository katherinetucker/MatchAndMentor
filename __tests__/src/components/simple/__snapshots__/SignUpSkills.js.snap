// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 28,
          }
        }
      />
    </View>
    <TextInput
      editable={true}
      maxLength={100}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Ex: JavaScript"
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "height": 40,
          "margin": 20,
        }
      }
      testID="skill-name-input"
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        disabled={undefined}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#55A838",
            "borderRadius": 10,
            "flex": 0.2,
            "height": 45,
            "justifyContent": "center",
            "margin": 20,
          }
        }
        testID={undefined}
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
            }
          }
        >
          1
        </Text>
      </TouchableHighlight>
      <TouchableHighlight
        activeOpacity={0.85}
        disabled={undefined}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#55A838",
            "borderRadius": 10,
            "flex": 0.2,
            "height": 45,
            "justifyContent": "center",
            "margin": 20,
          }
        }
        testID={undefined}
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
            }
          }
        >
          2
        </Text>
      </TouchableHighlight>
      <TouchableHighlight
        activeOpacity={0.85}
        disabled={undefined}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#55A838",
            "borderRadius": 10,
            "flex": 0.2,
            "height": 45,
            "justifyContent": "center",
            "margin": 20,
          }
        }
        testID={undefined}
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
            }
          }
        >
          3
        </Text>
      </TouchableHighlight>
      <TouchableHighlight
        activeOpacity={0.85}
        disabled={undefined}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#55A838",
            "borderRadius": 10,
            "flex": 0.2,
            "height": 45,
            "justifyContent": "center",
            "margin": 20,
          }
        }
        testID={undefined}
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
            }
          }
        >
          4
        </Text>
      </TouchableHighlight>
      <TouchableHighlight
        activeOpacity={0.85}
        disabled={undefined}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#55A838",
            "borderRadius": 10,
            "flex": 0.2,
            "height": 45,
            "justifyContent": "center",
            "margin": 20,
          }
        }
        testID={undefined}
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 24,
            }
          }
        >
          5
        </Text>
      </TouchableHighlight>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={undefined}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#333333",
          "borderRadius": 10,
          "flexWrap": "wrap",
          "height": 45,
          "justifyContent": "center",
          "margin": 5,
          "padding": 10,
        }
      }
      testID={undefined}
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 18,
          }
        }
      >
        test
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 30,
            "height": 30,
            "justifyContent": "center",
            "marginLeft": 10,
            "width": 30,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#000000",
            }
          }
        >
          1
        </Text>
      </View>
    </TouchableHighlight>
  </View>
  <View
    keyboardVerticalOffset={64}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#55A838",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="skills-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
